lms/services/digest.py

```diff
@@ -56,12 +75,20 @@ def send_instructor_email_digests(
                 # We don't have an email address for this user.
                 continue
 
-            self._mailchimp_service.send_template(
-                "instructor-email-digest",
-                self._sender,
-                recipient=EmailRecipient(to_email, unified_user.display_name),
-                template_vars=digest,
-            )
+            try:
+                self._mailchimp_service.send_template(
+                    "instructor-email-digest",
+                    self._sender,
+                    recipient=EmailRecipient(to_email, unified_user.display_name),
+                    template_vars=digest,
+                )
+            except MailchimpError as err:
+                errors[unified_user.h_userid] = err
+                LOG.exception(err)
+                report_exception(err)
```

Comment:
`errors` is a dict mapping the `h_userid`'s of the emails that failed to send to their `MailchimpError`'s. This is so that the Celery task can retry itself but only for the `h_userid`'s that failed, see https://github.com/hypothesis/lms/pull/5279/.

Then we also log the exception to Papertrail and report it to Sentry